builtin/k8s/platform.go

```diff
@@ -736,11 +752,234 @@ func (p *Platform) resourceDeploymentDestroy(
 	if err := deployclient.Delete(ctx, state.Name, metav1.DeleteOptions{}); err != nil {
 		return err
 	}
+
+	step.Update("Deployment deleted")
 	step.Done()
 
 	return nil
 }
 
+// resourceAutoscalerCreate creates the Kubernetes deployment.
+func (p *Platform) resourceAutoscalerCreate(
+	ctx context.Context,
+	log hclog.Logger,
+	src *component.Source,
+	img *docker.Image,
+	deployConfig *component.DeploymentConfig,
+	ui terminal.UI,
+
+	result *Deployment,
+	deployState *Resource_Deployment, // a deployment must exist before creating an autoscaler
+	state *Resource_Autoscale,
+	csinfo *clientsetInfo,
+	sg terminal.StepGroup,
+) error {
+	if p.config.AutoscaleConfig == nil {
+		log.Trace("no autoscale config detected, will not create one")
+		return nil
+	}
+
+	// Prepare our namespace and override if set.
+	ns := csinfo.Namespace
+	if p.config.Namespace != "" {
+		ns = p.config.Namespace
+	}
+
+	s := sg.Add("Preparing horizontal pod autoscaler...")
+	defer func() { s.Abort() }() // Defer in func in case more steps are added to this func in the future
+
+	clientSet := csinfo.Clientset
+	autoscaleClient := clientSet.AutoscalingV1().HorizontalPodAutoscalers(ns)
+
+	// Attempt to detect an existing metrics-server and display warning if none found
+	metricsServerLabel := "k8s-app=metrics-server"
+	metricsPods, err := clientSet.CoreV1().Pods("").List(ctx, metav1.ListOptions{
+		LabelSelector: metricsServerLabel,
+	})
+	if err != nil {
+		s.Update("Failed to list pods in attempt to detect existing metrics-server in cluster")
+		s.Status(terminal.StatusError)
+		s.Done()
+		return err
+	}
+
+	if len(metricsPods.Items) == 0 {
+		ui.Output("There were no pods recognized in the cluster as a metrics-server "+
+			"with the label '%s'. This means your autoscaler might "+
+			"not be functional until a metrics-server exists and can provide metrics to "+
+			"the horizontal pod autoscaler.",
```

Comment:
Is there a guide to installing a metric server we could link to from here? I guess it’d have to be a pretty dependable URL.

Reply:
Hmm, I tend to not like using live URLs to link people to something external if it's a domain we don't control... i.e. k8s docs change and we have to put out a release to fix the URL (or old clients get stuck with a broken URL for those who refuse to update).

Maybe we can wordsmith this to be clearer, i.e. give them enough of a nudge to know to look for how to install metrics-server.

Reply:
Yeah, as I user I know that I’d appreciate being led exactly to where I need to be to take the next step.

Reply:
I just pushed an additional sentence here, let me know what you think. This is tricky 😅

Reply:
@evanphx also brought up to include some kind of URL, so given there's a couple asks for it I added the github project URL!

Reply:
Seems great to me!